### PR TITLE
Unifed SQL message system

### PR DESCRIPTION
**Pull request by Jordie0608**
:cl:
tweak: Admin notes, memos and watchlist entries now use a generalized system, they can all be accessed from the former notes browser.
add: Added to this are messages, which allow admins to leave a message for players that is delivered to them when they next connect.
/:cl:

This replaces the separate handlings for notes, memos and the watchlist with a single system using generalized procs that write and read to the same table.
Added to this are 'messages' which allow admins to leave a message for players that is delivered to them when they next connect. The message is deleted after being sent. These are intended to supplant using the watchlist to ask other admins to PM a player when they seem them.
All of these are now accessed from the former notes browser. Notes, messages and watchlist entries are listed on player's pages; memos and a list of all watchlist entries have a page each to themselves. Existing note quick-links are preserved and memos are still shown to admins upon connection.

Also removes a debug line in minimaps that I noticed.

I'm pretty sure I tested every interaction but no guarantee I didn't miss something.

SQL changes will require MSO to be around upon merge.

### CODE REVIEW

**Comment by PKPenguin321** [comment]
>Added to this are messages, which allow admins to leave a message for players that is delivered to them when they next connect.

Fuck yeah, we should use this for people that suicide roundstart as a head

**Comment by Jordie0608** [comment]
Yes this was originally inspired by https://tgstation13.org/phpBB/viewtopic.php?f=29&t=9180

**Comment by PKPenguin321** [comment]
>linking to the secret board

**Comment by MrStonedOne** [comment]
Nothing stood out on a quick glance.

I'll have to see about pre-converting this and then making two way views in place of the old table to make things easier. Since otherwise sybil and bagil would have to be updated at once, and have their rounds restart at the same other or issues would happen.

**Comment by Jordie0608** [comment]
I'm sure if need be we could organize a situation where that'd happen =P

**Comment by optimumtact** [comment]
@MrStonedOne just let me know when you are ready.

The other option is to build an SQL update function into the codebase so snippets of sql can be run on roundstart (with the db version baked into a table somewhere)

File: SQL/database_changelog.txt [hunk 1]
```diff
@@ -1,3 +1,28 @@
...
+INSERT INTO `feedback`.`messages`
+(`id`,`type`,`targetckey`,`adminckey`,`text`,`timestamp`,`server`,`secret`,`lasteditor`,`edits`) SELECT `id`, "note", `ckey`, `adminckey`, `notetext`, `timestamp`, `server`, `secret`, `last_editor`, `edits` FROM `feedback`.`notes`
+
+INSERT INTO `feedback`.`messages`
+(`type`,`targetckey`,`adminckey`,`text`,`timestamp`,`lasteditor`,`edits`) SELECT "memo", `ckey`, `ckey`, `memotext`, `timestamp`, `last_editor`, `edits` FROM `feedback`.`memo`
+
+INSERT INTO `feedback`.`messages`
+(`type`,`targetckey`,`adminckey`,`text`,`timestamp`,`lasteditor`,`edits`) SELECT "watchlist entry", `ckey`, `adminckey`, `reason`, `timestamp`, `last_editor`, `edits` FROM `feedback`.`watch`
+
+It's not nessecary to delete the 'notes', 'memos', and 'watchlist' tables but they will no longer be used.
```
Comment (lzimann) [inline SQL/database_changelog.txt]: nessecary

**Comment by optimumtact** [comment]
@MrStonedOne @MrStonedOne @MrStonedOne @MrStonedOne @MrStonedOne @MrStonedOne @MrStonedOne @MrStonedOne @MrStonedOne @MrStonedOne @MrStonedOne @MrStonedOne @MrStonedOne @MrStonedOne @MrStonedOne @MrStonedOne @MrStonedOne @MrStonedOne @MrStonedOne @MrStonedOne @MrStonedOne @MrStonedOne @MrStonedOne @MrStonedOne @MrStonedOne @MrStonedOne @MrStonedOne @MrStonedOne @MrStonedOne @MrStonedOne @MrStonedOne @MrStonedOne @MrStonedOne @MrStonedOne @MrStonedOne @MrStonedOne @MrStonedOne @MrStonedOne @MrStonedOne @MrStonedOne @MrStonedOne @MrStonedOne @MrStonedOne @MrStonedOne

**Comment by MrStonedOne** [comment]
Well I'm not going to be able to do it the VIEW way because you can't insert into a view that has a static value like the memo/note/etc type shit.

I can still do it the view way but non-updated servers will not be able to insert new entries.

**Comment by Jordie0608** [comment]
Is it troublesome to set up whatever web-interface workbench like thing i vaguely remember you mentioning before or would you rather wait until the situation is apt for this merge?

**Comment by MrStonedOne** [comment]
Na, just hold up, I'll get this all ready and then we can merge+update.

**Comment by Poojawa** [comment]
would it be bad to ask for an idiot's guide to converting from the old notes, and being able to actually put new notes in?

**Comment by MrStonedOne** [comment]
from old old notes (file based)

or old notes (seperate mysql database)

**Comment by Jordie0608** [comment]
Which old notes? Pre-SQL notes: enable config https://github.com/tgstation/tgstation/blob/master/config/config.txt#L265 and the notes are auto-converted each time a player connects or uncomment this code block and run the proc once https://github.com/tgstation/tgstation/blob/master/code/modules/admin/sql_message_system.dm#L376

SQL notes: run the four commands listed here into your MySQL database https://github.com/tgstation/tgstation/blob/master/SQL/database_changelog.txt#L11

To add notes you open the messages interface and press the `Add Notes` button on either an index or player page.

**Comment by LetterJay** [comment]
The issue is when I add the notes, they're not being added. They were able to be moved from the old SQL table to the new message one. But if you try to add one in-game, nothing is added.

**Comment by Jordie0608** [comment]
It's functional on /tg/, check your server logs to see if there are any SQL errors being reported.

**Comment by LetterJay** [comment]
``SQL ERROR creating new note in messages table. Error : [Data too long for column 'server' at row 1]``

**Comment by optimumtact** [comment]
how long is your server name

**Comment by LetterJay** [comment]
Counting the description at the end? Around 41 characters.

**Comment by Jordie0608** [comment]
The `server` column is a max of 32 characters https://github.com/tgstation/tgstation/blob/master/SQL/tgstation_schema.sql#L345
either alter the column to a higher value or truncate your `config.server_name` value

**Comment by LetterJay** [comment]
Changing the char limit in the column fixed it. Thanks!

**Pull request merged**